src/validation_functions.jl

```diff
@@ -0,0 +1,132 @@
+mutable struct validation_info
+    field_descriptor::Union{Nothing, Dict}
+    struct_name::AbstractString
+    validation_action::Union{Nothing, Bool}
```

Comment:
I'm a bit confused on the type of validation_action. In the config files it is "warn" or "error" which essentially points to a function that will be executed.  Here it is stored as a boolean, which gets set to true on an error, but then I don't see it getting used anywhere.  Maybe I'm missing something.